src/main/java/io/confluent/kafkarest/v2/KafkaConsumerState.java

```diff
@@ -0,0 +1,425 @@
+		Pattern topicPattern = Pattern.compile(subscription.getTopicPattern());
+		NoOpOnRebalance noOpOnRebalance = new NoOpOnRebalance();
+		    consumer.subscribe(topicPattern, noOpOnRebalance);		
+	    }
+	}
+    } finally {
+      lock.writeLock().unlock();
+    }
+  }
+
+  public void unsubscribe() {
+    lock.writeLock().lock();
+    try {
+	if (consumer != null) {
+	    consumer.unsubscribe();
+	}
+    } finally {
+      lock.writeLock().unlock();
+    }
+  }
+
+  public java.util.Set<String> subscription() {
+    java.util.Set<String> currSubscription = null;
+    lock.writeLock().lock();
+    try {
+	if (consumer != null) {
+	    currSubscription = consumer.subscription();
+	}
+    } finally {
+      lock.writeLock().unlock();
+    }
+    return currSubscription;
+  }
+    
+
+  public java.util.Set<TopicPartition> assignment() {
+    java.util.Set<TopicPartition> currAssignment = null;
+    lock.writeLock().lock();
+    try {
+	if (consumer != null) {
+	    currAssignment = consumer.assignment();
+	}
+    } finally {
+      lock.writeLock().unlock();
+    }
+    return currAssignment;
+  }
+    
+
+  public ConsumerCommittedResponse committed(ConsumerCommittedRequest request) {
+    ConsumerCommittedResponse response  = new ConsumerCommittedResponse();
+    response.offsets = new Vector<TopicPartitionOffsetMetadata>();
+    lock.writeLock().lock();
+    try {
+	if (consumer != null) {
+	    for (io.confluent.kafkarest.entities.TopicPartition t: request.partitions) {
+		TopicPartition partition = new TopicPartition(t.getTopic(), t.getPartition());
+		OffsetAndMetadata offsetMetadata = consumer.committed(partition);
+		if (offsetMetadata != null) {
+		    response.offsets.add(new TopicPartitionOffsetMetadata(partition.topic(), partition.partition(), offsetMetadata.offset(), offsetMetadata.metadata()));
+		}
+	    }
+	}
+    } finally {
+      lock.writeLock().unlock();
+    }
+    return response;
+  }
+    
+    
+  public boolean expired(long nowMs) {
+    return expiration <= nowMs;
+  }
+
+  public void updateExpiration() {
+    this.expiration = config.getTime().milliseconds() +
+                      config.getInt(KafkaRestConfig.CONSUMER_INSTANCE_TIMEOUT_MS_CONFIG);
+  }
+
+  public long untilExpiration(long nowMs) {
+    return this.expiration - nowMs;
+  }
+
+  public KafkaRestConfig getConfig() {
+    return config;
+  }
+
+  public void setConfig(KafkaRestConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public int compareTo(KafkaConsumerState o) {
+    if (this.expiration < o.expiration) {
+      return -1;
+    } else if (this.expiration == o.expiration) {
+      return 0;
+    } else {
+      return 1;
+    }
+  }
+
+    public KafkaConsumerTopicState<KafkaK, KafkaV, ClientK, ClientV> getOrCreateTopicState(String topic, long timeout) {
+
+    lock.writeLock().lock();
+    try {
+      if (topics == null) {
+        return null;
+      }
+      if (topic != null) {
```

Comment:
This is not needed as topic is always null, and don't see the need for topic in V2 consumer